cfme/common/provider.py

```diff
@@ -198,7 +241,15 @@ def update(self, updates, cancel=False, validate_credentials=False):
         if not cancel:
             flash.assert_message_match('{} Provider "{}" was saved'.format(self.string_name, name))
 
+    @variable(alias='rest')
     def delete(self, cancel=True):
+        if cancel:
+            return
+        col = rest_api().collections.providers.find_by(name=self.name)[0]
```

Comment:
@lehinevych has a PR which has my commit tweaking the REST client a bit, including `.get()` which works similarly to `find_by` but it either returns the first item found or raises an exception. That can be used in these cases in the future.